### ISSUE TITLE
Package symbol confused with module

### ISSUE THREAD

**Issue by NeilGirdhar** [microsoft/pyright#4827]
The export [is here](https://github.com/pyqtgraph/pyqtgraph/blob/master/pyqtgraph/parametertree/__init__.py).

Yet, this code gives a type error:
```python
from pyqtgraph.parametertree import Parameter

def f(x: list[Parameter]) -> None:  # error: Module cannot be used as a type
    pass
```
I guess the import statement is setting `Parameter` due to the recent conversation here https://github.com/python/typing/issues/1358?  In any case, testing on the console seems to suggest that the symbol takes precedence.

**Comment by erictraut** [comment]
I presume that you have `useLibraryCodeForTypes` set to false. If you set it to true, then this resolves as you expect because pyright actually analyzes the library code and correctly determines that `pyqtgraph.parametertree.Parameter` is not just a module but is overwritten by the import loader by another symbol within `parametertree/__index__.py`. With `useLibraryCodeForTypes` set to false, pyright doesn't analyze the file but does resolve the submodule.

When `useLibraryCodeForTypes` is false, the type of the imported symbol `Parameter` should be `Unknown`. I'll work on a fix for this.

**Comment by erictraut** [comment]
This will be addressed in the next release.

**Comment by NeilGirdhar** [comment]
> I presume that you have `useLibraryCodeForTypes` set to false

I didn't even know about this feature!  Thank you for mentioning it :)

**Comment by erictraut** [comment]
I needed to revert my fix because it regressed some other cases. Will need to think more about this.

**Comment by NeilGirdhar** [comment]
Okay, no worries, thanks for the quick fixes!

**Comment by erictraut** [comment]
This is included in pyright 1.1.301, which I just published. It will also be included in a future release of pylance.
